_posts/en/newsletters/2023-01-11-newsletter.md

```diff
@@ -0,0 +1,130 @@
+    need to wait six blocks after they come online before they can use
+    that money to send a new payment over LN.
+
+    The authors suggest an alternative: user Alice chooses a
+    counterparty (Bob) in advance whose node she expects will always be
+    available (e.g. a Lightning Service Provider, LSP).  Alice and Bob
+    cooperate to create an onchain address for a script which allows
+    spending with a signature from Alice plus either a signature from
+    Bob or the expiration of a multi-week timelock (e.g. 6,000 blocks),
+    for [example][potentiam minsc]:
+
+    ```hack
+    pk(A) && (pk(B) || older(6000))
+    ```
+
+    This address can receive a payment that begins to accrue
+    confirmations while Alice is offline.  Until the payment has the
+    expiry number of confirmations, Bob must co-sign any attempt to
+    spend the money.  If Bob chooses to only sign a single spend attempt
+    that Alice also signs, then Bob can be sure that Alice can't double
+    spend that money before the expiry.  The only way Alice's spend
+    could become invalid is by the earlier payment to her also becoming
+    double spent.  If that payment has accrued many confirmations before
+    Alice came online and initiated her spend, a double spend should be
+    improbable.
+
+    This allows Alice to receive a payment while her wallet is offline,
+    come online after the payment has at least 6 confirmations (but
+    considerably less than 6,000 confirmations), and immediately co-sign
+    a transaction to open an LN channel that Bob knows can't be double
+    spent.  Even before that channel creation transaction is confirmed,
+    Bob can begin securely and trustlessly forwarding payments for
+    Alice.  Or, if Alice and Bob both already have LN channels (either
+    with each other or with separate peers), Bob can send an LN payment
+    to Alice which she can claim by spending her onchain funds to Bob.
+    Alternatively, if Alice's wallet comes online and she decides she
+    just wants to make a regular onchain payment, all she needs is for
+    Bob's to co-sign the spend.  In the worst case, if Bob becomes
+    uncooperative, Alice can simply wait a few weeks to spend her money
+    without his participation.
+
+    In addition to allowing offline wallets to receive funds for LN, the
+    authors describe how the idea might combine well with [async
+    payments][topic async payments] to allow LSPs to prepare channel
+    rebalance operations in advance for when an offline client comes
+    back online, allowing those rebalance operations to occur without
+    any delay from the user's perspective.  For example, if Carol sends
+    an async LN payment to Alice for an amount larger than the current
+    capacity in Alice's channel, Bob can send a payment to the script
+    `pk(B) && (pk(A) || older(6000)`.  If that payment has a sufficient
+    number of confirmations by the time Alice comes next comes online,
```

Comment:
If Bob sends money to Alice's address in this manner, isn't he just giving her money?

Reply:
This script is different than the script described earlier (the first was `A && (B || timeout)` ; the second is `B && (A || timeout)`.  I'll rephrase to try to make that clearer.

Reply:
Oh yeah, I totally missed that